swift/apple/FirezoneNetworkExtension/NetworkPathUpdates.swift

```diff
@@ -0,0 +1,23 @@
+import Network
+
+/// Creates an AsyncStream that emits NWPath updates.
+///
+/// Cancels the monitor via `onTermination` when the consuming Task is cancelled.
+/// This enables actors to manage NWPathMonitor lifecycle without needing
+/// `nonisolated(unsafe)` or `@unchecked Sendable`.
+func networkPathUpdates() -> AsyncStream<Network.NWPath> {
+  let (stream, continuation) = AsyncStream.makeStream(of: Network.NWPath.self)
+
+  let monitor = NWPathMonitor()
+  monitor.pathUpdateHandler = { path in
+    continuation.yield(path)
+  }
+  monitor.start(queue: .global())
```

Comment:
I am wondering if this should perhaps be an argument to the function? It might be good to know that we are spawning this on some kind of global executor?

Reply:
good point, I'll make it an argument with a default value to be more explicit.